### ISSUE TITLE
Bug in translating

### ISSUE THREAD

**Issue by Yevgnen** [jadore801120/attention-is-all-you-need-pytorch#28]
There's a mistake when repeating data for beam search.

The source seq [here](https://github.com/jadore801120/attention-is-all-you-need-pytorch/blob/master/transformer/Translator.py#L65)

```python
src_seq = Variable(src_seq.data.repeat(beam_size, 1))
```

gets a matrix in the following order for source sequence 

```
seq1
seq2
seq3
seq1
seq2
seq3
seq1
seq2
seq3
```

while the beam search input [here](https://github.com/jadore801120/attention-is-all-you-need-pytorch/blob/master/transformer/Translator.py#L82) 

```python
input_data = torch.stack([b.get_current_state() for b in beam if not b.done])
```

takes an input like

```
seq1
seq1
seq1
seq2
seq2
seq2
seq3
seq3
seq3
```
, both of which are fed into the decoder 

```python
            dec_outputs, dec_slf_attns, dec_enc_attns = self.model.decoder(
                input_data, input_pos, src_seq, enc_outputs)
```

The order of the two input does not match.

**Comment by hellozgy** [comment]
I find the same problem...

**Comment by jadore801120** [comment]
Hi @Yevgnen and @hellozgy.
This bug has been fixed in commit [defe385](https://github.com/jadore801120/attention-is-all-you-need-pytorch/commit/defe385d68ec9c36c2fc8148b9fe58a4ea059337).
Thanks for pointing out!

Yu-Hsiang
